src/test/java/org/jenkinsci/plugins/workflow/cps/CpsScmFlowDefinitionTest.java

```diff
@@ -195,6 +202,28 @@ public class CpsScmFlowDefinitionTest {
         r.assertLogContains("version one", b);
     }
 
+    @Issue("JENKINS-42971")
+    @Test
+    public void lightweight_brach_parametrised() throws Exception {
+        sampleRepo.init();
+        sampleRepo.git("checkout","-b","master2");
+        sampleRepo.write("flow.groovy", "echo 'version one'");
+        sampleRepo.git("add", "flow.groovy");
+        sampleRepo.git("commit", "--message=init");
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.addProperty(new ParametersDefinitionProperty(new StringParameterDefinition("BRANCH","","")));
+        GitSCM scm = new GitSCM(GitSCM.createRepoList(sampleRepo.toString(), null),
+                new ArrayList<BranchSpec>() {{
+                    add(new BranchSpec("${BRANCH}"));
+                }}, null, null, Collections.emptyList());
+
+        CpsScmFlowDefinition def = new CpsScmFlowDefinition(scm, "flow.groovy");
+        def.setLightweight(true);
+        p.setDefinition(def);
+
+        r.assertBuildStatusSuccess(p.scheduleBuild2(0, new ParametersAction(new StringParameterValue("BRANCH","master2"))));
```

Comment:
For reference, what is the behavior with the `git` bump locally reverted?

Reply:
> For reference, what is the behavior with the `git` bump locally reverted?

It fails as expected (as before)
`
22.135 [p #1] hudson.plugins.git.GitException: Command "git.exe fetch --tags --force --progress --prune -- origin +refs/heads/${BRANCH}:refs/remotes/origin/${BRANCH}" returned status code 128:
`
because the BRANCH isnt expanded. But it is backward compatible if I specify master2 directly it passes.